src/world/types.rs

```diff
@@ -51,7 +51,47 @@ impl<'lua> FromLua<'lua> for ModuleInputs<'lua> {
     }
 }
 
-pub type Location = [f64; 3];
+pub struct Location(pub f64, pub f64, pub f64, pub f64, pub f64, pub f64);
+
+impl<'lua> FromLua<'lua> for Location {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> Result<Self> {
```

Comment:
I had to convert Location from a fixed-size array to a tuple. You can't implement traits for types that someone else wrote. A tuple is considered a custom type, while an array is more of an alias I guess; it's not my own type that I can implement traits for.

Reply:
And now it's a struct with named fields instead of numbers. `l.angle_velocity` is much better than `l.5`.